client/signup/controller.js

```diff
@@ -295,6 +296,11 @@ export default {
 		if ( ! isEmpty( additionalDependencies ) ) {
 			context.store.dispatch( updateDependencies( additionalDependencies ) );
 		}
+		if ( themeParameter ) {
+			setSignupSelectedThemeSlug( query.theme );
```

Comment:
I think it would be better to set the query string via the `getChecklistThemeDestination` function because that's why we store the `themeParameter` in `additionalDependencies` (L290 - L294)

See https://github.com/Automattic/wp-calypso/blob/bac42aae3c2c623f5028c7498416267dd83551cb/client/signup/config/flows.js#L115

Reply:
Besides, I think it would be better to create a new flow and include only the PA step. It's a bit weird to redirect the user to a certain step of the flow after the signup. What do you think?

For example, [podcast flow](https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/podcasts.ts#L8), [free flow](https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/free.ts#L28), etc

Reply:
> create a new flow and include only the PA step

That would be extremely helpful for https://github.com/Automattic/wp-calypso/pull/72286 too!